lumicks/pylake/tests/data/mock_confocal.py

```diff
@@ -186,31 +211,268 @@ def make_slice(data):
         )
 
 
-def generate_kymo(name, image, pixel_size_nm, start=4, dt=7, samples_per_pixel=5, line_padding=3):
-    confocal_file, metadata, stop = MockConfocalFile.from_image(
+def generate_kymo(
+    name,
+    image,
+    pixel_size_nm=10.0,
+    start=20e9,
+    dt=62.5e9,
+    samples_per_pixel=5,
+    line_padding=3,
+    with_ref=False,
+):
+    return generate_confocal(
+        name,
         image,
+        scan=False,
         pixel_sizes_nm=[pixel_size_nm],
-        axes=[0],
-        start=np.int64(start),
-        dt=np.int64(dt),
+        start=start,
+        dt=dt,
         samples_per_pixel=samples_per_pixel,
         line_padding=line_padding,
+        with_ref=with_ref,
+        multi_color=image.ndim > 2,
     )
 
-    return Kymo(name, confocal_file, start, stop, metadata)
-
 
 def generate_scan(
-    name, scan_data, pixel_sizes_nm, start=4, dt=7, samples_per_pixel=5, line_padding=3
+    name,
+    image,
+    pixel_sizes_nm,
+    axes=None,
+    start=20e9,
+    dt=62.5e9,
+    samples_per_pixel=5,
+    line_padding=3,
+    with_ref=False,
+    multi_color=False,
+):
+    return generate_confocal(
+        name,
+        image,
+        scan=True,
+        pixel_sizes_nm=pixel_sizes_nm,
+        axes=axes,
+        start=start,
+        dt=dt,
+        samples_per_pixel=samples_per_pixel,
+        line_padding=line_padding,
+        with_ref=with_ref,
+        multi_color=multi_color,
+    )
+
+
+def generate_confocal(
+    name,
+    image,
+    scan=False,
+    pixel_sizes_nm=[10.0],
+    axes=None,
+    start=20e9,
+    dt=62.5e9,
+    samples_per_pixel=5,
+    line_padding=3,
+    with_ref=False,
+    multi_color=False,
 ):
+    start = np.int64(start)
+    dt = np.int64(dt)
+    axes = (np.arange(min(2, image.ndim)) if scan else [0]) if axes is None else axes
+
     confocal_file, metadata, stop = MockConfocalFile.from_image(
-        np.swapaxes(scan_data, -1, -2),
+        image,
         pixel_sizes_nm=pixel_sizes_nm,
-        axes=np.arange(scan_data.ndim),
-        start=np.int64(start),
-        dt=np.int64(dt),
+        axes=axes,
+        start=start,
+        dt=dt,
+        samples_per_pixel=samples_per_pixel,
+        line_padding=line_padding,
+        multi_color=multi_color,
+    )
+
+    confocal_class = Scan if scan else Kymo
+    confocal = confocal_class(name, confocal_file, start, stop, metadata)
+
+    if not with_ref:
+        return confocal
+
+    # Assume the x axis to be the slow and the y axis to be the fast indexed one: The x axis
+    # contains the `lines_per_frame` and the y axis the `pixels_per_line`:
+    number_of_frames, pixels_per_line, lines_per_frame = (
+        [1, *image.shape[:2]] if len(image.shape) == 2 + multi_color else image.shape[:3]
+    )
+
+    x_axis_fast = len(axes) == 2 and axes[0] < axes[1]
```

Comment:
Could also be Y axis fast (if the second axis is Z).

Reply:
Right, what do you think about `first_physcial_axis_fast`? It's long but descriptive.

Reply:
sounds good